Duplicati/Library/Utility/CallContextSettings.cs

```diff
@@ -277,11 +276,11 @@ public static T Settings
         /// </summary>
         /// <returns>The disposable handle for the context.</returns>
         /// <param name="initial">The initial value.</param>
-        public static IDisposable StartContext(string settingsKey, T initial = default(T))
+        public static IDisposable StartContext(string settingsKey, T? initial = default(T))
         {
             contextSettingsType = settingsKey;
             var res = new ContextGuard();
-            Settings = initial;
+            Settings = initial ?? throw new ArgumentNullException(nameof(initial), "The initial value cannot be null.");

```

Comment:
This is a contradiction. `default(T)` is null for any reference type.

Reply:
Made a fox for this as well